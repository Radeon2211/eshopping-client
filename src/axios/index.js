import axios from 'axios';
import adapter from 'axios/lib/adapters/http';

const serverURL = 'https://radeon2211-eshopping.herokuapp.com';
const localURL = 'http://192.168.1.109:4000';

export const baseURL = serverURL;

const instance = axios.create({
  baseURL,
  withCredentials: true,
  adapter,
});

const getCsrfToken = async () => {
  const { data } = await instance.get('/csrf-token');
  instance.defaults.headers.post['X-CSRF-Token'] = data.csrfToken;
};
getCsrfToken();

export default instance;
