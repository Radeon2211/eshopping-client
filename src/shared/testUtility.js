// eslint-disable-next-line import/no-extraneous-dependencies
import checkPropTypes from 'check-prop-types';

export const checkProps = (component, expectedProps) => {
  // eslint-disable-next-line react/forbid-foreign-prop-types
  return checkPropTypes(component.propTypes, expectedProps, 'props', component.name);
};

export const historyPageNum = (pageNumber) => ({
  listen: jest.fn(),
  location: { search: `?p=${pageNumber}` },
  createHref: jest.fn(),
  push: jest.fn(),
});

export const propsPagination = (itemQuantity = 5) => ({
  itemQuantity,
  isDataLoading: false,
  maxQuantityPerPage: 2,
});
