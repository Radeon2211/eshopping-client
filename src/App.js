/* eslint-disable react/jsx-wrap-multilines */
import React, { useCallback, useEffect, Suspense, lazy } from 'react';
import { Route, Redirect, Switch } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import axios from './axios';
import * as actions from './store/actions/indexActions';
import { DEFAULT_PATH } from './shared/constants';
import theme from './styled/theme';

import Heading from './components/UI/Heading/Heading';
import Modal from './components/UI/Modal/Modal';
import MessageBox from './components/UI/MessageBox/MessageBox';
import Navbar from './components/Navbar/Navbar';
import Loader from './components/UI/Loader';
import Main from './components/UI/Main';
import Footer from './components/Footer';
import Products from './containers/Products';
import Logout from './containers/Logout';

const ProductDetails = lazy(() => import('./containers/ProductDetails/ProductDetails'));
const OtherUser = lazy(() => import('./containers/OtherUser/OtherUser'));
const MyAccount = lazy(() => import('./containers/MyAccount/MyAccount'));
const Cart = lazy(() => import('./containers/Cart/Cart'));
const Transaction = lazy(() => import('./containers/Transaction/Transaction'));

const WaitingComponent = (Component) => {
  return (props) => (
    <Suspense
      fallback={
        <div style={{ textAlign: 'center' }}>
          <Loader size="big" />
        </div>
      }
    >
      <Component {...props} />
    </Suspense>
  );
};

const App = () => {
  const userProfile = useSelector((state) => state.auth.profile);

  const dispatch = useDispatch();
  const onFetchProfile = useCallback(() => dispatch(actions.fetchProfile()), [dispatch]);

  useEffect(() => {
    const getCsrfToken = async () => {
      try {
        const { data } = await axios.get('/csrf-token');
        axios.defaults.headers.post['X-CSRF-Token'] = data.csrfToken;
        onFetchProfile();
      } catch (error) {
        onFetchProfile();
      }
    };
    getCsrfToken();
  }, [onFetchProfile]);

  let routes = (
    <div style={{ textAlign: 'center', marginTop: theme.spacings.level3 }}>
      <Loader size="big" />
    </div>
  );

  if (userProfile === null) {
    if (!axios.defaults.headers.post['X-CSRF-Token']) {
      routes = (
        <Heading variant="h3" align="center" mgTop="level3">
          Server connection error. Try again later
        </Heading>
      );
    } else {
      routes = (
        <>
          <Modal />
          <MessageBox />
          <Navbar userProfile={userProfile} />
          <Main>
            <Switch>
              <Route path="/products/:id" exact component={WaitingComponent(ProductDetails)} />
              <Route path="/products" exact component={Products} />
              <Redirect to="/products" />
            </Switch>
          </Main>
          <Footer />
        </>
      );
    }
  }
  if (userProfile) {
    routes = (
      <>
        <Modal />
        <MessageBox />
        <Navbar userProfile={userProfile} />
        <Main>
          <Switch>
            <Route path="/logout" component={Logout} />
            <Route path="/products/:id" exact component={WaitingComponent(ProductDetails)} />
            <Route path="/products" exact component={Products} />
            <Route path="/user/:username" exact component={WaitingComponent(OtherUser)} />
            <Route path="/cart" exact component={WaitingComponent(Cart)} />
            <Route path="/transaction" exact component={WaitingComponent(Transaction)} />
            <Route path="/my-account" component={WaitingComponent(MyAccount)} />
            <Redirect to={DEFAULT_PATH} />
          </Switch>
        </Main>
        <Footer />
      </>
    );
  }

  return routes;
};

export default App;
