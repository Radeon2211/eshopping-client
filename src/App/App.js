import React, { useCallback, useEffect, Suspense, lazy } from 'react';
import { Route, Routes, useLocation, Navigate } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { ErrorBoundary } from 'react-error-boundary';
import axios from '../axios';
import * as actions from '../store/actions/indexActions';
import { defaultAppPath, userStatuses } from '../shared/constants';
import ErrorPage from '../components/UI/ErrorPage/ErrorPage';
import FlexWrapper from '../components/UI/FlexWrapper';
import { ReactComponent as ServerErrorImage } from '../images/server-connection-error.svg';
import { ReactComponent as UnexpectedBugImage } from '../images/unexpected-bug.svg';

import Modal from '../components/UI/Modal/Modal';
import MessageBox from '../components/UI/MessageBox/MessageBox';
import Navbar from '../components/Navbar/Navbar';
import Loader from '../components/UI/Loader/Loader';
import Main from '../components/UI/Main';
import Footer from '../components/Footer/Footer';
import Products from '../containers/Products/Products';
import LandingPage from '../containers/LandingPage/LandingPage';
import MyData from '../containers/MyAccount/MyData/MyData';
import MyProducts from '../containers/MyAccount/MyProducts/MyProducts';
import MySellHistory from '../containers/MyAccount/MySellHistory/MySellHistory';
import MyPlacedOrders from '../containers/MyAccount/MyPlacedOrders/MyPlacedOrders';

const OrderDetails = lazy(() => import('../containers/OrderDetails/OrderDetails'));
const ProductDetails = lazy(() => import('../containers/ProductDetails/ProductDetails'));
const OtherUser = lazy(() => import('../containers/OtherUser/OtherUser'));
const MyAccount = lazy(() => import('../containers/MyAccount/MyAccount'));
const Cart = lazy(() => import('../containers/Cart/Cart'));
const Transaction = lazy(() => import('../containers/Transaction/Transaction'));
const Logout = lazy(() => import('../containers/Logout/Logout'));

const WaitingComponent = (Component) => {
  return (props) => (
    <Suspense
      fallback={
        <div style={{ textAlign: 'center' }}>
          <Loader size="big" />
        </div>
      }
    >
      <Component {...props} />
    </Suspense>
  );
};

const App = () => {
  const { pathname } = useLocation();

  const userProfile = useSelector((state) => state.auth.profile);

  const dispatch = useDispatch();
  const onFetchProfile = useCallback(() => dispatch(actions.fetchProfile()), [dispatch]);

  useEffect(() => {
    const getCsrfToken = async () => {
      try {
        const { data } = await axios.get('/csrf-token');
        axios.defaults.headers.post['X-CSRF-Token'] = data.csrfToken;
        onFetchProfile();
      } catch (error) {
        onFetchProfile();
      }
    };
    getCsrfToken();
  }, [onFetchProfile]);

  let content = (
    <FlexWrapper $justify="center" $mgTop="3" data-testid="App-loader">
      <Loader size="big" />
    </FlexWrapper>
  );
  let routes = null;

  if (userProfile === null) {
    if (!axios.defaults.headers.post['X-CSRF-Token']) {
      content = (
        <ErrorPage
          info="Oops! Server connection error. Please try again later"
          icon={<ServerErrorImage />}
        />
      );
    } else {
      routes = (
        <Main data-testid="App-user-null">
          <Routes>
            <Route path="/user/:username" Component={WaitingComponent(OtherUser)} />
            <Route path="/product/:id" Component={WaitingComponent(ProductDetails)} />
            <Route path="/products" Component={Products} />
            <Route path="/" Component={LandingPage} />
            <Route path="*" element={<Navigate to={defaultAppPath} replace />} />
          </Routes>
        </Main>
      );
    }
  } else if (userProfile) {
    if (userProfile.status === userStatuses.ACTIVE) {
      routes = (
        <Main data-testid="App-user-active">
          <Routes>
            <Route path="/logout" Component={WaitingComponent(Logout)} />
            <Route path="/order/:id" Component={WaitingComponent(OrderDetails)} />
            <Route path="/product/:id" Component={WaitingComponent(ProductDetails)} />
            <Route path="/products" Component={Products} />
            <Route path="/user/:username" Component={WaitingComponent(OtherUser)} />
            <Route path="/cart" Component={WaitingComponent(Cart)} />
            <Route path="/transaction" Component={WaitingComponent(Transaction)} />
            <Route path="/my-account" Component={WaitingComponent(MyAccount)}>
              <Route path="data" Component={MyData} />
              <Route path="products" element={<MyProducts userProfile={userProfile} />} />
              <Route path="sell-history" Component={MySellHistory} />
              <Route path="placed-orders" Component={MyPlacedOrders} />
              <Route path="*" element={<Navigate to="/my-account/data" replace />} />
            </Route>
            <Route path="*" element={<Navigate to={defaultAppPath} replace />} />
          </Routes>
        </Main>
      );
    } else {
      routes = (
        <Main data-testid="App-user-pending">
          <Routes>
            <Route path="/logout" Component={WaitingComponent(Logout)} />
            <Route path="/product/:id" Component={WaitingComponent(ProductDetails)} />
            <Route path="/products" Component={Products} />
            <Route path="/user/:username" Component={WaitingComponent(OtherUser)} />
            <Route path="/my-account" Component={WaitingComponent(MyAccount)}>
              <Route path="data" Component={MyData} />
              <Route path="*" element={<Navigate to="/my-account/data" replace />} />
            </Route>
            <Route path="*" element={<Navigate to={defaultAppPath} replace />} />
          </Routes>
        </Main>
      );
    }
  }

  if (routes) {
    content = (
      <>
        <Modal />
        <MessageBox />
        {pathname !== '/' && <Navbar userProfile={userProfile} />}
        {routes}
        {pathname !== '/' && <Footer />}
      </>
    );
  }

  return (
    <ErrorBoundary
      fallback={
        <ErrorPage
          info="Something went wrong. Please refresh the page"
          icon={<UnexpectedBugImage />}
        />
      }
    >
      {content}
    </ErrorBoundary>
  );
};

export default App;
