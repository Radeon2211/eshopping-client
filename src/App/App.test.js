import React from 'react';
import { screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import axios from '../axios';
import App from './App';
import { defaultUserProfile, renderAppPart } from '../shared/testUtility/testUtility';
import * as actions from '../store/actions/indexActions';
import { userStatuses } from '../shared/constants';

const mockStore = configureMockStore([thunk]);

const setUp = (userProfile, ui = { message: '', isFormLoading: false }, pathname = '/logout') => {
  const store = mockStore({
    auth: {
      profile: userProfile,
    },
    ui,
  });
  store.dispatch = jest.fn();

  return {
    ...renderAppPart(<App />, {
      pathname,
      store,
    }),
    store,
  };
};

jest.mock('../store/actions/indexActions.js', () => ({
  fetchProfile: () => {},
  logoutUser: () => {},
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

describe('<App />', () => {
  const axiosMock = new MockAdapter(axios);

  beforeEach(() => {
    axiosMock.onGet('/csrf-token').reply(200, {
      csrfToken: 'token',
    });
    jest.spyOn(console, 'error').mockImplementation(() => {});
    delete axiosMock.axiosInstance.defaults.headers.post['X-CSRF-Token'];
  });

  describe('check how renders', () => {
    it('should render version for user with active status', () => {
      setUp(defaultUserProfile);
      expect(screen.getByTestId('App-user-active')).toBeInTheDocument();
    });

    it('should render version for user with pending status', () => {
      setUp({
        ...defaultUserProfile,
        status: userStatuses.PENDING,
      });
      expect(screen.getByTestId('App-user-pending')).toBeInTheDocument();
    });

    it('should render server connection error if we do NOT wait for setting token in useEffect()', () => {
      setUp(null);
      expect(axiosMock.axiosInstance.defaults.headers.post['X-CSRF-Token']).toBeUndefined();
      expect(
        screen.getByText('Oops! Server connection error. Please try again later'),
      ).toBeInTheDocument();
    });

    it('should render loading screen', () => {
      setUp(undefined);
      expect(screen.getByTestId('App-loader')).toBeInTheDocument();
    });

    it('should render error from ErrorBoundary if data from redux ui reducer are not available', () => {
      setUp(defaultUserProfile, null);
      expect(screen.getByText('Something went wrong. Please refresh the page')).toBeInTheDocument();
      // expect(screen.getByTestId('App-error-boundary')).toBeInTheDocument();
    });
  });

  describe('check useEffect()', () => {
    it('should set token in useEffect() after waiting', async () => {
      setUp(null);
      expect(axiosMock.axiosInstance.defaults.headers.post['X-CSRF-Token']).toBeUndefined();
      await waitFor(() => {
        expect(axiosMock.axiosInstance.defaults.headers.post['X-CSRF-Token']).toEqual('token');
      });
    });

    it('should call fetchProfile() and axios.get(/csrf-token) successfully', async () => {
      const { store } = setUp(undefined);

      expect(axiosMock.axiosInstance.defaults.headers.post['X-CSRF-Token']).toBeUndefined();
      expect(store.dispatch).not.toHaveBeenCalled();
      await waitFor(() => {
        expect(store.dispatch).toHaveBeenCalledWith(actions.fetchProfile());
      });
      await waitFor(() => {
        expect(axiosMock.axiosInstance.defaults.headers.post['X-CSRF-Token']).toEqual('token');
      });
    });

    it('should call fetchProfile() successfully and axios.get(/csrf-token) NOT successfully', async () => {
      axiosMock.onGet('/csrf-token').reply(500);

      const { store } = setUp(undefined);

      expect(store.dispatch).not.toHaveBeenCalled();
      await waitFor(() => {
        expect(store.dispatch).toHaveBeenCalledWith(actions.fetchProfile());
      });
      await waitFor(() => {
        expect(axiosMock.axiosInstance.defaults.headers.post['X-CSRF-Token']).toBeUndefined();
      });
    });
  });
});
