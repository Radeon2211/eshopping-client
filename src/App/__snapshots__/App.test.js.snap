// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> check how renders should render error from ErrorBoundary 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-top: 2.4rem;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > *:not(:last-child) {
  margin-bottom: 2.4rem;
}

.c2 {
  font-size: 2.4rem;
  font-family: Noto Sans,sans-serif;
  line-height: 1;
  margin-bottom: 2.4rem;
  margin-top: 0;
  text-align: center;
  line-height: 1.3;
}

.c1 {
  padding: 0 1.2rem;
}

.c1 > svg {
  max-width: 100%;
  width: 60rem;
}

@media only screen and (max-width:75em) {

}

@media only screen and (max-width:56.25em) {

}

@media only screen and (max-width:37.5em) {

}

@media only screen and (max-width:37.5em) {

}

@media only screen and (max-width:56.25em) {

}

@media only screen and (max-width:37.5em) {

}

<div
    class="c0 c1"
    direction="column"
    spacing="3"
    width="100%"
    wrap="nowrap"
  >
    <h3
      class="c2"
    >
      Something went wrong. Please refresh the page
    </h3>
    <svg>
      unexpected-bug.svg
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<App /> check how renders should render loading screen 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 2.4rem;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > *:not(:last-child) {
  margin-right: 0;
}

.c1 {
  display: inline-block;
  position: relative;
  width: 7rem;
  height: 7rem;
}

.c1 > div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  border-radius: 50%;
  -webkit-animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  width: 7rem;
  height: 7rem;
  border: 0.8rem solid #1E8BC3;
  border-color: #1E8BC3 transparent transparent transparent;
}

.c1 > div:nth-child(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.c1 > div:nth-child(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c1 > div:nth-child(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

<div
    class="c0"
    direction="row"
    spacing=""
    width="100%"
    wrap="nowrap"
  >
    <div
      class="c1"
      data-testid="Loader"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<App /> check how renders should render server connection error if we do NOT wait for setting token in useEffect() 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  margin-top: 2.4rem;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 > *:not(:last-child) {
  margin-bottom: 2.4rem;
}

.c2 {
  font-size: 2.4rem;
  font-family: Noto Sans,sans-serif;
  line-height: 1;
  margin-bottom: 2.4rem;
  margin-top: 0;
  text-align: center;
  line-height: 1.3;
}

.c1 {
  padding: 0 1.2rem;
}

.c1 > svg {
  max-width: 100%;
  width: 60rem;
}

<div
    class="c0 c1"
    direction="column"
    spacing="3"
    width="100%"
    wrap="nowrap"
  >
    <h3
      class="c2"
    >
      Oops! Server connection error. Please try again later
    </h3>
    <svg>
      server-connection-error.svg
    </svg>
  </div>
</DocumentFragment>
`;

exports[`<App /> check how renders should render version for user with active status 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Lobster,cursive;
  font-size: 3.3rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-shadow: 0 0 3px rgba(0,0,0,0.3);
  margin-top: 0;
  text-align: left;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 > svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  fill: #777;
  width: 3.4rem;
  height: 3.4rem;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c9 > svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  fill: #777;
  width: 1.7rem;
  height: 1.7rem;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

.c8 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0.8rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  text-align: right;
  font-size: 1.5rem;
  max-width: 15rem;
}

.c3 {
  box-shadow: 0 0 3px rgba(0,0,0,0.03);
  border-radius: 1px;
  cursor: pointer;
  font-size: 1.4rem;
  font-weight: 700;
  line-height: 1.2;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  outline: none;
  padding: 0.8rem 1.2rem;
  position: relative;
  text-align: center;
  text-transform: uppercase;
  -webkit-transition: all 0.1s;
  transition: all 0.1s;
  background-color: #1E8BC3;
  border: 2px solid #1E8BC3;
  color: #fff;
}

.c3:hover {
  background-color: #22A7F0;
  border-color: #22A7F0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FAFAFA;
  box-shadow: 0 1px 5px rgba(0,0,0,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  padding: 0.8rem 3.6rem;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 10000;
}

.c0 > *:not(:last-child) {
  margin-right: 3.6rem;
}

.c0 .logo {
  width: 4.6rem;
  height: 4.6rem;
  margin-right: 0.8rem;
}

.c0 .header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 100%;
  width: 45rem;
}

.c2 .name-input {
  background-color: #FAFAFA;
  border: 1px solid #BFBFBF;
  font-size: 1.5rem;
  outline: none;
  padding: 0.8rem;
  width: 100%;
}

.c5 {
  position: relative;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 > *:not(:last-child) {
  margin-right: 3.6rem;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.8rem 0;
  position: relative;
}

@media only screen and (max-width:75em) {
  .c0 .heading {
    display: none;
  }
}

@media only screen and (max-width:56.25em) {
  .c0 {
    padding: 0.8rem 2.4rem;
  }

  .c0 > *:not(:last-child) {
    margin-right: 2.4rem;
  }
}

@media only screen and (max-width:37.5em) {
  .c0 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }

  .c0 > *:nth-child(2) {
    margin-right: 0;
    margin-top: 0.8rem;
    width: 100%;
  }
}

@media only screen and (max-width:37.5em) {
  .c2 {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media only screen and (max-width:56.25em) {
  .c4 > *:not(:last-child) {
    margin-right: 2.4rem;
  }
}

@media only screen and (max-width:37.5em) {

}

<header
    class="c0"
  >
    <a
      data-testid="Navbar-header-link"
    >
      <header
        class="header"
      >
        <img
          alt="E-Shopping"
          class="logo"
          src="logo.png"
        />
        <h1
          class="c1 heading"
        >
          shopping
        </h1>
      </header>
    </a>
    <form
      class="c2"
    >
      <input
        autocomplete="off"
        class="name-input"
        data-testid="SearchForm-input"
        maxlength="150"
        name="product"
        spellcheck="false"
        type="text"
        value=""
      />
      <button
        aria-label="Search"
        class="c3"
        color="blue"
        type="submit"
      >
        search
      </button>
    </form>
    <nav
      class="c4"
      data-testid="LoggedInLinks"
    >
      <a
        class="c5"
        data-testid="CartLink"
      >
        <div
          class="c6"
          color=""
        >
          <svg>
            cart.svg
          </svg>
        </div>
      </a>
      <div
        class="c7"
        data-testid="LoggedInLinks-user-box"
        id="user"
      >
        <span
          class="c8"
          color=""
          display="inline-block"
          overflow="hidden"
          size="3"
          spacing="0px"
          transform=""
        >
          username
        </span>
        <div
          class="c9"
          color=""
        >
          <svg>
            arrow.svg
          </svg>
        </div>
        <div />
      </div>
    </nav>
  </header>
  .c1 {
  display: inline-block;
  position: relative;
  width: 7rem;
  height: 7rem;
}

.c1 > div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  border-radius: 50%;
  -webkit-animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  animation: rotate 1.2s cubic-bezier(0.5,0,0.5,1) infinite;
  width: 7rem;
  height: 7rem;
  border: 0.8rem solid #1E8BC3;
  border-color: #1E8BC3 transparent transparent transparent;
}

.c1 > div:nth-child(1) {
  -webkit-animation-delay: -0.45s;
  animation-delay: -0.45s;
}

.c1 > div:nth-child(2) {
  -webkit-animation-delay: -0.3s;
  animation-delay: -0.3s;
}

.c1 > div:nth-child(3) {
  -webkit-animation-delay: -0.15s;
  animation-delay: -0.15s;
}

.c0 {
  margin: 0 auto;
  max-width: 120rem;
  min-height: 100vh;
  padding: 8.7rem 1.2rem 2.4rem;
}

@media only screen and (max-width:75em) {

}

@media only screen and (max-width:56.25em) {

}

@media only screen and (max-width:37.5em) {

}

@media only screen and (max-width:37.5em) {

}

@media only screen and (max-width:56.25em) {

}

@media only screen and (max-width:37.5em) {
  .c0 {
    padding-top: 13.6rem;
  }
}

<main
    class="c0"
    data-testid="App-user-active"
  >
    <div
      style="text-align: center;"
    >
      <div
        class="c1"
        data-testid="Loader"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </main>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 1.2rem;
  margin-right: 0;
  margin-bottom: 1.2rem;
  margin-left: 0;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 > *:not(:last-child) {
  margin-right: 2.4rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 > svg {
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  fill: #777;
  width: 2.5rem;
  height: 2.5rem;
}

.c1 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.4rem;
}

.c4 {
  display: inline-block;
  font-style: normal;
  font-weight: normal;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
  margin-bottom: 0;
  margin-left: 0.8rem;
  margin-right: 0;
  overflow: visible;
  text-overflow: clip;
  white-space: normal;
  font-size: 1.3rem;
}

.c0 {
  border-top: 1px solid #BFBFBF;
  padding: 2.4rem 1.2rem;
  text-align: center;
}

.c0 .social-link {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 .about-website {
  color: #777;
  cursor: pointer;
  font-size: 1.3rem;
  -webkit-transition: color 0.1s;
  transition: color 0.1s;
}

.c0 .about-website:hover {
  color: #000;
}

@media only screen and (max-width:75em) {

}

@media only screen and (max-width:56.25em) {

}

@media only screen and (max-width:37.5em) {

}

@media only screen and (max-width:37.5em) {

}

@media only screen and (max-width:56.25em) {

}

@media only screen and (max-width:37.5em) {

}

<footer
    class="c0"
  >
    <span
      class="c1"
      color=""
      display="inline-block"
      overflow="visible"
      size="2"
      spacing="0px"
      transform=""
    >
      © 2021 Radosław Mikrut. All rights reserved
    </span>
    <div
      class="c2"
      direction="row"
      spacing="3"
      width="100%"
      wrap="nowrap"
    >
      <a
        class="social-link"
        href="https://www.linkedin.com/in/rados%C5%82aw-mikrut-a8600b198/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="c3"
          color=""
        >
          <svg>
            linkedin.svg
          </svg>
        </div>
      </a>
      <a
        class="social-link"
        href="mailto:radoslawmikrut@wp.pl"
        rel="noopener noreferrer"
      >
        <div
          class="c3"
          color=""
        >
          <svg>
            email.svg
          </svg>
        </div>
        <span
          class="c4"
          color=""
          display="inline-block"
          overflow="visible"
          size="1"
          spacing="0px"
          transform=""
        >
          radoslawmikrut@wp.pl
        </span>
      </a>
    </div>
    <span
      class="about-website"
      role="button"
      tabindex="0"
    >
      About website
    </span>
  </footer>
</DocumentFragment>
`;
